### ISSUE TITLE
MD022 - Headings should be surrounded by blank lines

### ISSUE THREAD

**Issue by walles** [DavidAnson/markdownlint#143]
I'd like to configure this warning to:
* Require a preceding empty line (as it already does)
* Not requiring a following empty line

So this should be OK:
``` markdown
something

# Heading
more text
```

But this should not be OK:
``` markdown
something
# Heading
more text
```

[The docs](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md022) explain why you must have an empty line before the heading (kramdown), but only cites aesthetics for why you should have an empty line after the heading.

And as a lot of our markdown has no empty line after headings, and I actually like it this way, I'd like to be able to inform Markdownlint about this.

  Regards /Johan

**Comment by DavidAnson** [comment]
I feel like I have seen parsers have problems with this, so I won’t default it on - but it seems okay to add as an option. Thanks!

**Comment by vukanac** [comment]
Is it possible to configure markdownlinter to force usage of 2 empty lines before headings and 1 after?

**Comment by DavidAnson** [comment]
Not currently, but that seems like a reasonable thing to enable as part of addressing this issue.

**Comment by vukanac** [comment]
Thank you for new feature!
But new problem show up. Applying this new rule will force to change other rule (multiple empty lines), and that will allow breaking its desired default value 1. As in `.markdownlint.json`:
```json
{
  "no-multiple-blanks": {
    "maximum": 2         <------ We need to keep this to 1.
  },
  "blanks-around-headings": {
    "line-above": 2,
    "line-below": 1
  }
}
```
Desired would be only to override multiple-empty-line validation on validating heading lines.
Any way, thank you for the work on this feature.

**Comment by DavidAnson** [comment]
Yes, I mention that consequence here: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md022

Because rules are independent, I don’t see a good way to avoid this.

**Comment by merqrial** [comment]
Related question, can we have heading-level rules for this? Probably makes sense to have a blank line after the H1, not necessarily headings lower than H4.

**Comment by DavidAnson** [comment]
See #504
